eng/pipelines/templates/steps/az-login.yml

```diff
@@ -16,10 +16,11 @@ steps:
       az account set `
         --subscription "$($subscriptionConfiguration.SubscriptionId)"
 
-            # Export service principal auth information for terraform testing
-            Write-Host "##vso[task.setvariable variable=arm-client-id;issecret=true]$($subscriptionConfiguration.TestApplicationId)"
-      Write-Host "##vso[task.setvariable variable=arm-client-secret;issecret=true]$($subscriptionConfiguration.TestApplicationSecret)"
-      Write-Host "##vso[task.setvariable variable=arm-tenant-id;issecret=true]$($subscriptionConfiguration.TenantId)"
+      # Export service principal auth information for terraform testing
+      Write-Host "##vso[task.setvariable variable=ARM_CLIENT_ID;]$($subscriptionConfiguration.TestApplicationId)"
+      Write-Host "##vso[task.setvariable variable=ARM_CLIENT_SECRET;issecret=true]$($subscriptionConfiguration.TestApplicationSecret)"
```

Comment:
Isn't is required for secret variables to be passed to script tasks via `env`?

Reply:
@weikanglim by default, setting a secret variable via task.setvariable doesn't also set it as an environment variable. In my latest push (after your comment) I fix this by calling task.setvariable a second time for ARM_CLIENT_SECRET without `issecret=true`.

Reply:
There seem to be some issues with my understanding of this, reverting to the original method.